### PR TITLE
Meta: Enforce .tsx extensions

### LINKED ISSUE #4158: Source link in extension settings doesn't work for features with `.ts` extension
As an example, `collapse-markdown-sections` feature links to:
`https://github.com/sindresorhus/refined-github/blob/main/source/features/collapse-markdown-sections.tsx`
rather than:
`https://github.com/sindresorhus/refined-github/blob/main/source/features/collapse-markdown-sections.ts`

### PR DESCRIPTION
**Pull request by yakov116**
Resolves #4158

### CODE REVIEW

**Comment by fregante** [comment]
Looks like this copy is repeated from elsewhere. Let’s make a generic throwError function that throws or logs a regular single-line emoji-less string

**Comment by yakov116** [comment]
What would that be different then just using onsole.error?

We are not adding anything. Unless you have something else in mind.

File: webpack.config.ts [hunk 1]
```diff
@@ -114,7 +114,24 @@ const config: Configuration = {
...
+
+						❌ Feature \`${fileInfo.name}\` has a ${fileInfo.ext} extension but should be .tsx
+
+						`;
+						if (isWatching) {
+							console.error(error);
+							return JSON.stringify(fileInfo.name);
+						}
+
+						throw new Error(error);
```
[the remark this answers is not in the document]
Comment (yakov116) [reply webpack.config.ts:130]: The issue is that you are throwing during watching. Which means that until you add a readme the watch will crash.

Please correct me if I am wrong.

Or do you want me to do a is watching check there?

**Review by fregante — APPROVED** [review]
Untested but it looks so good it should run as Miss America 2021

File: webpack.config.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (fregante) [reply webpack.config.ts]: You solved it correctly

**Pull request merged**